zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneService.java

```diff
@@ -88,13 +171,305 @@ private ConnectionDetails buildConnectionDetails(String zoneName, IdentityZone i
+        int firstDotIndex = domainName.indexOf('.');
+        if (firstDotIndex == -1)  return "";
+        return domainName.substring(firstDotIndex);
+    }
+
+
+    private Map<String, String> getKeys(String zoneSigningKey, String activeKeyId) {
+        Map<String, String> keysMap = new HashMap<>();
+        Map<String, Map<String, String>> keys = new HashMap<>();
+        Map<String, String> signingKeyMap = new HashMap<>();
+        signingKeyMap.put("signingKey", zoneSigningKey);
+        String keysStr = JsonUtils.writeValueAsString(signingKeyMap);
+        keysMap.put(activeKeyId, keysStr);
+        return keysMap;
+    }
+
+    private String createSigningKey() throws NoSuchAlgorithmException, IOException {
+
+        StringWriter pemStringWriter = new StringWriter();
+        JcaPEMWriter pemWriter = new JcaPEMWriter(pemStringWriter);
+
+        KeyPairGenerator keyPairGenerator = null;
+        keyPairGenerator = KeyPairGenerator.getInstance("RSA");
+        keyPairGenerator.initialize(2048);
+        pemWriter.writeObject(keyPairGenerator.genKeyPair().getPrivate());
+        pemWriter.flush();
+        pemWriter.close();
+        return pemStringWriter.toString();
+    }
+
+    private SamlConfig createSamlConfig(String subdomain) throws IOException, NoSuchAlgorithmException, OperatorCreationException {
+        JcePEMEncryptorBuilder builder = new JcePEMEncryptorBuilder("DES-EDE3-CBC");
+        builder.setProvider("BC");
+        String passphrase = new RandomValueStringGenerator(8).generate();
+        PEMEncryptor pemEncryptor = builder.build(passphrase.toCharArray());
+        KeyPairGenerator rsa = KeyPairGenerator.getInstance("RSA");
+        rsa.initialize(2048);
+        KeyPair kp = rsa.generateKeyPair();
+
+        JcaMiscPEMGenerator pemGenerator = new JcaMiscPEMGenerator(kp.getPrivate(), pemEncryptor);
+        StringWriter pemStringWriter = new StringWriter();
+        JcaPEMWriter pemWriter = new JcaPEMWriter(pemStringWriter);
+        pemWriter.writeObject(pemGenerator);
+        pemWriter.flush();
+        pemWriter.close();
+
+        Calendar cal = Calendar.getInstance();
+        cal.add(Calendar.YEAR, 10);
+
+        byte[] pk = kp.getPublic().getEncoded();
+        SubjectPublicKeyInfo bcPk = SubjectPublicKeyInfo.getInstance(pk);
+        String dn = "C=US, ST=CA, L=San Ramon, O=GE, OU=GE Digital, CN=PredixUAA"+subdomain;
+        X509v1CertificateBuilder certGen = new X509v1CertificateBuilder(
+            new X500Name(dn),
+            BigInteger.ONE,
+            new Date(),
+            cal.getTime(),
+            new X500Name(dn),
+            bcPk
+        );
+        X509CertificateHolder certHolder = certGen
+            .build(new JcaContentSignerBuilder("SHA256withRSA").build(kp.getPrivate()));
+
+        HashMap<String, SamlKey> samlKeys = new HashMap<>();
+
+        String certificate = BEGIN_CERT + "\n" + base64encoder.encode(certHolder.getEncoded()) + "\n" + END_CERT;
```

Comment:
encode method returns byte array, better wrap it up with to string like,

Arrays.toString(base64encoder.encode(certHolder.getEncoded()))

Reply:
Done. Thanks!